  * Add support for auto.master in LDAP.  Make autofs-ldap depend on
    ldap-utils to find the automount LDAP schemas.  Patch from Rune
    Skillingstad. (Closes: #193028)

Index: debian/control
===================================================================
RCS file: /var/lib/cvs/skolelinux/src/rebuilds/autofs/debian/control,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -3 -p -u -r1.1.1.1 -r1.2
--- debian/control	21 Sep 2003 08:34:27 -0000	1.1.1.1
+++ debian/control	21 Sep 2003 10:06:26 -0000	1.2
@@ -25,6 +25,6 @@ Description: Hesiod map support for auto
 
 Package: autofs-ldap
 Architecture: any
-Depends: ${shlibs:Depends}, autofs
+Depends: ${shlibs:Depends}, autofs, ldap-utils
 Description: LDAP map support for autofs
  This package provides support for LDAP lookup entries within autofs.
Index: debian/default
===================================================================
RCS file: /var/lib/cvs/skolelinux/src/rebuilds/autofs/debian/default,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -3 -p -u -r1.1.1.1 -r1.2
--- debian/default	21 Sep 2003 08:34:27 -0000	1.1.1.1
+++ debian/default	21 Sep 2003 10:06:26 -0000	1.2
@@ -1,2 +1,6 @@
 # Timeout value in seconds (default: 300)
 TIMEOUT=300
+# LDAP server
+LDAPSERVER=ldap
+# LDAP dn for auto.master
+MASTERDN=ou=auto.master,ou=Automount,dc=skole,dc=skolelinux,dc=no
Index: debian/patches/043_ldap_master
===================================================================
RCS file: debian/patches/043_ldap_master
diff -N debian/patches/043_ldap_master
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ debian/patches/043_ldap_master	21 Sep 2003 10:06:26 -0000	1.1
@@ -0,0 +1,52 @@
+diff -ur autofs-4.0.0pre10.orig/samples/rc.autofs.in autofs-4.0.0pre10/samples/rc.autofs.in
+--- autofs-4.0.0pre10.orig/samples/rc.autofs.in	Sun Sep 21 11:35:23 2003
++++ autofs-4.0.0pre10/samples/rc.autofs.in	Sun Sep 21 11:54:18 2003
+@@ -89,7 +89,9 @@
+ {
+     if ! getmounts_file /etc/auto.master; then
+ 	if ! getnismounts auto.master; then
+-	    echo "no autofs mounts configured!" 1>&2
++	    if ! getldapmounts; then
++		echo "no autofs mounts configured!" 1>&2
++	    fi
+ 	fi
+     fi
+ }
+@@ -229,6 +231,35 @@
+     else
+ 	return 1
+     fi
++}
++
++function getldapmounts() {
++    if [ ! -x /usr/bin/ldapsearch -o -z $LDAPSERVER -o -z $MASTERDN ]; then
++	return 1
++    fi   
++    ldapsearch -LLL -h $LDAPSERVER -b $MASTERDN -x \
++	'(objectClass=automount)' cn automountInformation 2>/dev/null \
++	| while read attr val; do
++	      case "$attr" in
++  	          dn:)
++                      node=""
++		      args=""
++		      ;;
++	          cn:)
++		      node=$val
++		      ;;
++		  automountInformation:)
++		      args=$val
++		      ;;
++		  *)
++		      ;;
++	      esac
++	      if [ -n "$node" -a -n "$args" ]; then
++		  echo "$node $args" | process_master_file "$@"
++		  node=""
++		  args=""
++	      fi
++	  done
+ }
+ 
+ #
+Only in autofs-4.0.0pre10.orig/samples: rc.autofs.in.orig
+Only in autofs-4.0.0pre10/samples: rc.autofs.in~
